src/network/connection.cpp

```diff
@@ -89,31 +101,80 @@ bool Connection::endPacket() {
 	return r > 0;
 }
 
+bool Connection::beginBundle() {
+	if (!m_ServerFeatures.has(ServerFeatures::PROTOCOL_BUNDLE_SUPPORT)) {
+		return false;
+	}
+
+	if (!m_Connected || m_IsBundle) {
+		return false;
+	}
+
+	if (beginPacket()) {
+		sendPacketType(PACKET_BUNDLE);
+		sendPacketNumber();
+		m_IsBundle = true;
+		return true;
+	}
+	
+	return false;
+}
+
+bool Connection::endBundle() {
+	m_IsBundle = false;
+
+	if (!m_ServerFeatures.has(ServerFeatures::PROTOCOL_BUNDLE_SUPPORT)) {
+		return false;
+	}
```

Comment:
```suggestion
	if(!m_IsBundle)
	    return;
```

Reply:
```suggestion
	if (!m_IsBundle) {
		return false;
	}
	
	m_IsBundle = false;
```
Still need to clear the flag before ending that packet